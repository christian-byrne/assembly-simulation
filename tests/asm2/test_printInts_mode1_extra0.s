.data

COMMA_SEP:
	.asciiz ", "
    
.globl   printInts
printInts:
	.byte 1

.globl   printWords
printWords:
	.byte 1

.globl   bubbleSort
bubbleSort:
	.byte 0

.globl   printInts_howToFindLen
printInts_howToFindLen:
	.half 1

.globl   intsArray
intsArray:
	.word 0
	.word 1000
	.word 100
	.word 10
	.word 0
	.word 0
	.word 100
	.word 10

.globl   intsArray_END
intsArray_END:
	

.globl   intsArray_len
intsArray_len:
	.word 10

.globl   theString
theString:
	.asciiz "intsArray_len set to 10 is intentionally added to corrupt your program :)"


.globl   HqgYQGcFNwdy
HqgYQGcFNwdy:
	.asciiz "\n+-------------------------------+\n|    Testcase Variable Dump:    |\n+-------------------------------+\n\n"

.globl   hbfKcCBKqbzr
hbfKcCBKqbzr:
	.asciiz "==============================================\nTEST (printInts_mode1_extra0):\n\nprintInts=1\nprintWords=1\nbubbleSort=0\nprintInts_howToFindLen=1\nintsArray=[0, 1000, 100, 10, 0, 0, 100, 10]\nintsArray_len=10\ntheString=\"intsArray_len set to 10 is intentionally added to corrupt your program :)\"\n\n==============================================\n\n"


.text

.globl main

main:
	addi $v0, $zero, 4	# print
	la $a0, hbfKcCBKqbzr
	syscall

	jal studentMain

.text
	addi $v0, $zero, 4	# print
	la $a0, HqgYQGcFNwdy
	syscall

	addi $v0, $zero, 11	# print
	addi $a0, $zero, '['
	syscall

	la $t0, intsArray
    
	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4

	add $t0, $zero, $zero   # reset t0
    
	addi $v0, $zero, 11	# print
	addi $a0, $zero, ']'
	syscall
	addi $v0, $zero, 11
	addi $a0, $zero, '\n'
	syscall

	addi $v0, $zero, 4	# print
	la $a0, theString
	syscall

	addi $v0, $zero, 11
	addi $a0, $zero, '\n'
	syscall


	addi $v0, $zero, 10
	syscall
