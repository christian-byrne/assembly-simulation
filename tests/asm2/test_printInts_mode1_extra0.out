==============================================
TEST (printInts_mode1_extra0):

printInts=1
printWords=1
bubbleSort=0
printInts_howToFindLen=1
intsArray=[0, 1000, 100, 10, 0, 0, 100, 10]
intsArray_len=10
theString="intsArray_len set to 10 is intentionally added to corrupt your program :)"

==============================================

printInts: About to print 8 elements.
0
1000
100
10
0
0
100
10
printWords: There were 12 words.
:)
program
your
corrupt
to
added
intentionally
is
10
to
set
intsArray_len

+-------------------------------+
|    Testcase Variable Dump:    |
+-------------------------------+

[0, 1000, 100, 10, 0, 0, 100, 10]
intsArray_len
